core/src/main/java/org/elasticsearch/index/query/BaseTermQueryBuilder.java

```diff
@@ -146,18 +146,18 @@ public QueryValidationException validate() {
     }
 
     @Override
-    protected final int doHashCode() {
-        return Objects.hash(getClass(), fieldName, value);
+    protected int doHashCode() {
+        return Objects.hash(fieldName, value);
```

Comment:
I don't think removing getClass is a good move, that means that a span term query and a term query with same field and same value would be equal